### PR TITLE
search by name func with test

### PR DESCRIPTION
**Pull request by reemsh2000**
Relates #1

### CODE REVIEW

**Comment by yasmeenattallah** [comment]
please add a new line  at the end of each file

File: Js/test.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
+const searchByName = require('./main.js');
+const data =require('./data.js')
+test('should return all the objects contain the pass name',()=>{
+    const actual=searchByName('Formal')
```
Comment (yasmeenattallah) [inline Js/test.js:4]: you want to test search by name function  but you pass 'formal' which is a category of an item, not the name

File: Js/test.js [hunk 2]
```diff
@@ -0,0 +1,23 @@
+const searchByName = require('./main.js');
+const data =require('./data.js')
```
Comment (yasmeenattallah) [inline Js/test.js:2]: you don't need to require data here

File: Js/main.js [hunk 3]
```diff
@@ -0,0 +1,13 @@
+const data =require('./data.js')
+// const searchInput=document.querySelector('.search-box')
+//Search By Name
+// searchInput.addEventListener('keypress', searchByName(searchInput.value))
+const searchByName =((search)=>{
+  const SearchedArray= data.filter((ele)=>{
```
Comment (yasmeenattallah) [inline Js/main.js:6]: filter function actually returns the elements that pass the condition so you don't need an if statement here just pass the condition like this 
```   const SearchedArray= data.filter((ele)=> ele.name ===  search   ```

File: Js/main.js [hunk 4]
```diff
@@ -0,0 +1,13 @@
+const data =require('./data.js')
+// const searchInput=document.querySelector('.search-box')
+//Search By Name
+// searchInput.addEventListener('keypress', searchByName(searchInput.value))
+const searchByName =((search)=>{
```
Comment (yasmeenattallah) [inline Js/main.js:5]: it's better to  replace 'search' parameter with 'searchedName' to be more conventional

**Review by asmaaamin — APPROVED** [review]

**Review by mohammedsalah7 — APPROVED** [review]

**Pull request merged**